lib/daru/core/query.rb

```diff
@@ -70,6 +70,23 @@ def vector_where dv, bool_array
           resultant_dv
         end
 
+        def vector_apply_where dv, bool_array, &block
+          _data, new_index = fetch_new_data_and_index dv, bool_array
+          all_index = dv.index.to_a
+          yield_data = dv.map(&block).to_a
+          all_data = all_index.map { |idx| new_index.include?(idx) ? yield_data[idx] : dv[idx] }
```

Comment:
Are you sure it is a sane approach (apply block to all data, and then select appropriate)? 
Imagine there is 10k values vector, and "where" returns only 5 values of all 10k, and calculation in `block` is expensive (some serious math, or even reading some files). "Right" implementation should call block 5 times, not 10k times and then select only 5 values.